ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -1091,7 +1091,7 @@ public boolean checkLeadership() {
    * 4. If majority respond success, returns readIndex.
    * @return current readIndex.
    */
-  CompletableFuture<Long> getReadIndex(Long readAfterWriteConsistentIndex) {
+  CompletableFuture<Long> getReadIndex(RaftServerConfigKeys.Read.Option option, Long readAfterWriteConsistentIndex) {
```

Comment:
It seems that with lease read we need to change the name of this function (and maybe some other functions on the call stack). The important logic for both readIndex and leaseRead is three steps.
1. Verify that the current node is the Leader (readIndex uses a logical clock to initiate heartbeat, lease read uses a physical clock with an error upper bound),
2. Wait for the local applyIndex to reach the current commitIndex
3. Query the state machine.

Maybe we could name the first step confirmLeader or something

Reply:
They both need to confirm leader **and obtain readIndex** first. ReadOnly(ReadIndex based) uses majority heartbeats to obtain the readIndex while ReadOnly(Lease based) uses leader lease. I think readIndex is a shared step

Reply:
Got it! This really depends on how to understand the meaning of the word `readIndex`. If we consider that the follower also needs to get commitIndex from the leader when using lease read, it is reasonable for me to support current naming logic, so please ignore this review~